nrtk_explorer/app/embeddings.py

```diff
@@ -123,18 +125,23 @@ def on_run_transformations(self, transformed_image_ids):
             )
 
         elif self.state.tab == "UMAP":
-            self.state.points_transformations = self.reducer.reduce(
+            points_transformations = self.reducer.reduce(
                 name="UMAP",
                 fit_features=self.features,
                 features=transformation_features,
                 dims=self.state.dimensionality,
             )
 
+        self.state.points_transformations = points_transformations[
+            len(self.state.points_sources or []) :
```

Comment:
I undertsand this workaround, after merging this I will modify the dimension_reducer class to handle this.